httpx/client.py

```diff
@@ -238,6 +259,31 @@ def _get_auth_middleware(
 
         return None
 
+    def _dispatcher_for_request(
+        self, request: AsyncRequest, proxies: typing.Dict[str, AsyncDispatcher]
+    ) -> AsyncDispatcher:
+        """Gets the AsyncDispatcher instance that should be used for a given Request"""
+        if proxies:
+            url = request.url
+            is_default_port = (url.scheme == "http" and url.port == 80) or (
+                url.scheme == "https" and url.port == 443
+            )
+            hostname = f"{url.host}:{url.port}"
+            proxy_keys = (
+                f"{url.scheme}://{hostname}",
+                f"{url.scheme}://{url.host}" if is_default_port else None,
+                f"all://{hostname}",
+                f"all://{url.host}" if is_default_port else None,
+                url.scheme,
+                "all",
+            )
+            for proxy_key in proxy_keys:
+                if proxy_key and proxy_key in proxies:
+                    dispatcher = proxies[proxy_key]
+                    return dispatcher
+
+        return self.dispatch
```

Comment:
Is this logic covered in the tests? Seems like there should be some for the prioritization of `all` over specific IPs and the host defaulting.

Reply:
All kinda just means "catch-all" in this scenario. We follow requests lead here.